cpp/src/barretenberg/stdlib/primitives/logic/logic.test.cpp

```diff
@@ -19,81 +31,139 @@ template <class T> void ignore_unused(T&) {} // use to ignore unused variables i
+
+using ComposerTypes =
+    ::testing::Types<honk::StandardHonkComposer, plonk::StandardComposer, plonk::TurboComposer, plonk::UltraComposer>;
+
+TYPED_TEST_SUITE(LogicTest, ComposerTypes);
+
+TYPED_TEST(LogicTest, TestCorrectLogic)
+{
+    STDLIB_TYPE_ALIASES
+
+    auto run_test = [](size_t num_bits, Composer& composer) {
+        uint256_t mask = (uint256_t(1) << num_bits) - 1;
+
+        uint256_t a = engine.get_random_uint256() & mask;
+        uint256_t b = engine.get_random_uint256() & mask;
+
+        uint256_t and_expected = a & b;
+        uint256_t xor_expected = a ^ b;
+
+        field_ct x = witness_ct(&composer, a);
+        field_ct y = witness_ct(&composer, b);
+
+        field_ct x_const(&composer, a);
+        field_ct y_const(&composer, b);
+
+        field_ct and_result = stdlib::logic<Composer>::create_logic_constraint(x, y, num_bits, false);
+        field_ct xor_result = stdlib::logic<Composer>::create_logic_constraint(x, y, num_bits, true);
+
+        field_ct and_result_left_constant =
+            stdlib::logic<Composer>::create_logic_constraint(x_const, y, num_bits, false);
+        field_ct xor_result_left_constant =
+            stdlib::logic<Composer>::create_logic_constraint(x_const, y, num_bits, true);
+
+        field_ct and_result_right_constant =
+            stdlib::logic<Composer>::create_logic_constraint(x, y_const, num_bits, false);
+        field_ct xor_result_right_constant =
+            stdlib::logic<Composer>::create_logic_constraint(x, y_const, num_bits, true);
+
+        field_ct and_result_both_constant =
+            stdlib::logic<Composer>::create_logic_constraint(x_const, y_const, num_bits, false);
+        field_ct xor_result_both_constant =
+            stdlib::logic<Composer>::create_logic_constraint(x_const, y_const, num_bits, true);
+
+        EXPECT_EQ(uint256_t(and_result.get_value()), and_expected);
+        EXPECT_EQ(uint256_t(and_result_left_constant.get_value()), and_expected);
+        EXPECT_EQ(uint256_t(and_result_right_constant.get_value()), and_expected);
+        EXPECT_EQ(uint256_t(and_result_both_constant.get_value()), and_expected);
+
+        EXPECT_EQ(uint256_t(xor_result.get_value()), xor_expected);
+        EXPECT_EQ(uint256_t(xor_result_left_constant.get_value()), xor_expected);
+        EXPECT_EQ(uint256_t(xor_result_right_constant.get_value()), xor_expected);
+        EXPECT_EQ(uint256_t(xor_result_both_constant.get_value()), xor_expected);
+    };
+
+    auto composer = Composer();
+    for (size_t i = 8; i < 248; i += 8) {
+        run_test(8, composer);
+    }
+    auto prover = composer.create_prover();
+    plonk::proof proof = prover.construct_proof();
+    auto verifier = composer.create_verifier();
+    bool result = verifier.verify_proof(proof);
+    EXPECT_EQ(result, true);
+}
+
+// Tests the constraints will fail if the operands are larger than expected even though the result contains the correct
```

Comment:
I would actually like some opinions on what the behaviour should be if we provide `a` and `b` to be larger than `num_bits` wdyt?

Reply:
I think the XOR/AND should be the truncation of the XOR/AND of the two inputs, and the and the proof should pass. Thoughts?

Reply:
Ok, so we should  shorten `a` and `b` to `num_bits` and then applying the range constraint?